feat: Add nursing tasks

ChillarAnand opened the pull request:
Docs: https://docs.erpnext.com/docs/v13/user/manual/en/nursing_tasks

From checking vitals to ensuring medications are administered to a patient in a timely manner, nursing tasks are essential part of healthcare workflows. Frappe Healthcare ensures nursing tasks are transitioned correctly through out the workflows.

Creating nursing checklist template.

<img width="1280" alt="Screenshot 2021-11-08 at 11 11 49" src="https://user-images.githubusercontent.com/4463796/140690265-bad5ea0d-e186-4769-b7d7-da4aee51ea37.png">

Link nursing checklist to lab test template.

<img width="1280" alt="Screenshot 2021-11-08 at 11 13 04" src="https://user-images.githubusercontent.com/4463796/140690277-24e0fc7b-584d-40ea-aba6-f0bbe58cec1f.png">

Create lab test from template and it should create nursing tasks.

<img width="1280" alt="Screenshot 2021-11-08 at 11 13 20" src="https://user-images.githubusercontent.com/4463796/140690278-6b1d02bd-8f11-471e-8437-69b57106cf67.png">

Users cant submit lab test before completing nursing tasks.

<img width="1280" alt="Screenshot 2021-11-08 at 11 13 39" src="https://user-images.githubusercontent.com/4463796/140690279-3b503cd9-d845-4b34-a894-1f6c95b40728.png">

To see linked nursing tasks click nursing tasks button.

<img width="1280" alt="Screenshot 2021-11-08 at 11 13 48" src="https://user-images.githubusercontent.com/4463796/140690286-cf6748dd-7c41-4365-8881-7d2680cf9bf4.png">

To modify nursing task on a case to case basis, click on `modify nursing tasks` and select required tasks.

<img width="1274" alt="Screenshot 2021-11-25 at 18 37 50" src="https://user-images.githubusercontent.com/4463796/143447227-5437c403-4af8-4c5d-a94d-5f9b7f80ea14.png">

<img width="1274" alt="Screenshot 2021-11-25 at 18 38 02" src="https://user-images.githubusercontent.com/4463796/143447253-fbf8e1d0-9769-4d5e-a14d-5b559c66a64a.png">

ChillarAnand committed:
feat: Added models for nursing tasks

ChillarAnand committed:
feat: Added child table for nursing tasks

ChillarAnand committed:
feat: Auto create nursing tasks for labs

ChillarAnand committed:
feat: Added missing fields for task

ChillarAnand committed:
feat: Clean up therapy plan

ChillarAnand committed:
feat: Added template for therapy plan

ChillarAnand committed:
feat: Create nursing tasks for clinical procedure

ChillarAnand committed:
feat: Added validation for nursing tasks

ChillarAnand committed:
feat: Added validation for nursing tasks

ChillarAnand committed:
feath: Add checks for nursing task validation

ChillarAnand committed:
feat: Add nursing tasks for therapy sessions

ChillarAnand committed:
feat: Added checklist for admission & discharge

ChillarAnand committed:
feat: Provide modal to select nursing tasks

ChillarAnand committed:
Merge branch 'develop' into nursing-tasks

ruchamahabal commented:
@ChillarAnand Can you update the PR description with why the feature was added and the flow of creation?

ChillarAnand commented:
> @ChillarAnand Can you update the PR description with why the feature was added and the flow of creation?

Added docs for the same at https://docs.erpnext.com/docs/v13/user/manual/en/nursing_tasks. @ruchamahabal

ChillarAnand committed:
feat: Clean up nursing tasks tests

ChillarAnand committed:
feat: Add tests for modifying nursing tasks

ruchamahabal commented on healthcare/healthcare/doctype/nursing_checklist_template/nursing_checklist_template.json:
```diff
@@ -0,0 +1,65 @@
+   "fieldname": "title",
+   "fieldtype": "Data",
+   "label": "Title",
+   "unique": 1
```
Title should be mandatory

ruchamahabal commented on healthcare/healthcare/doctype/healthcare_activity/healthcare_activity.json:
```diff
@@ -0,0 +1,43 @@
+ ],
+ "sort_field": "modified",
+ "sort_order": "DESC",
+ "track_changes": 1
```
Enable Quick Entry for this doctype so that it can be created from any view without navigating to that form.

ruchamahabal replied:
Set auto naming for this doctype based on the activity field
![image](https://user-images.githubusercontent.com/24353136/149080909-f549ea74-e811-44bb-9bb8-d8c8d6cba987.png)

akurungadam replied:
added title as name field and enabled quick entry

ruchamahabal commented on healthcare/healthcare/doctype/nursing_checklist_template/nursing_checklist_template.json:
```diff
@@ -0,0 +1,65 @@
+  },
+  {
+   "fieldname": "tasks",
+   "fieldtype": "Table",
```
Redundant labeling for table and section? Just keep the section label

![image](https://user-images.githubusercontent.com/24353136/149080275-9b8cf481-4aab-471b-8467-ac0ab27e81d3.png)

ruchamahabal replied:
Layout in Child Table and a few pointers:

![image](https://user-images.githubusercontent.com/24353136/149082026-5441ff5d-0376-4e33-8257-1a1f8b49c845.png)

- This table should be mandatory
- Description should not be in grid view since it can be opened for more context. Most nurses would be familiar with the title of the activity
- This field should be a part of Activity doctype instead since it can be reused in multiple templates and users won't keep filling it up for each. Fetch desc from the activity and let the user edit it if need be.
- **Starts On** needs to be an Int field. It is meant for the day on which this should be done relative to the date of the transaction. Since this template will be used in multiple transactions same date won't be applicable for all. Check [Employee Onboarding Template](https://docs.erpnext.com/docs/v13/user/manual/en/human-resources/employee-onboarding#31-employee-onboarding-template) for context.
- Following fields should be shown in Grid View instead:
	- Activity
	- Type
	- Starts On
	- Duration
	- Mandatory
	- Role
- Add a column break in the child table layout to unclutter the fields.
- Add more context for the Task DocType field. Labeling should be Reference Document Type. You can add a description for this field: Reference Document to be Validated (eg: If reference is Vital Signs system will validate whether vital signs were actually captured during the nursing task)

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.json:
```diff
@@ -555,14 +556,21 @@
  "is_submittable": 1,
  "links": [],
- "modified": "2020-11-30 11:04:17.195848",
+ "modified": "2021-11-29 06:16:23.354524",
```
For some reason, the patient is getting reset in Lab Test. Looks like it's trying to fetch from some field.

![Kapture 2022-01-12 at 13 06 55](https://user-images.githubusercontent.com/24353136/149084091-47d36635-b3a7-4d4c-a614-410d4f758f3f.gif)

akurungadam replied:
couldn't see this happening, could you please check once again?

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.py:
```diff
@@ -19,6 +20,8 @@ def on_submit(self):
 		self.validate_result_values()
 		self.db_set('submitted_date', getdate())
 		self.db_set('status', 'Completed')
+		from healthcare.healthcare.utils import validate_nursing_tasks
```
We can keep all the imports together?

akurungadam replied:
this is to get around a circular import, will fix this as part of the lab module redesign

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.js:
```diff
@@ -57,9 +57,49 @@ frappe.ui.form.on('Lab Test', {
-	}
+		frm.add_custom_button(__("Nursing Tasks"), function () {
+				frappe.set_route("List", "Nursing Task", {"reference_name": frm.doc.name});
+		});
```
This button should not be shown before saving:

![image](https://user-images.githubusercontent.com/24353136/149084322-84795526-f917-4793-ab11-58bb70693bf8.png)

Also, club it under a parent button View -> Nursing Tasks. Since just "Nursing Tasks" is ambiguous.

akurungadam replied:
moved the custom buttons to Connections

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.js:
```diff
@@ -57,9 +57,49 @@ frappe.ui.form.on('Lab Test', {
-	}
+		frm.add_custom_button(__("Nursing Tasks"), function () {
+				frappe.set_route("List", "Nursing Task", {"reference_name": frm.doc.name});
+		});
```
```suggestion
		frm.add_custom_button(__("Nursing Tasks"), function () {
			frappe.set_route("List", "Nursing Task", {"reference_name": frm.doc.name});
		});
```

Over indented?

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.js:
```diff
@@ -57,9 +57,49 @@ frappe.ui.form.on('Lab Test', {
+				});
+			}
+		});
+	},
```
- Mixed tabs and spaces:
- Also try to use uniform quote styles throughout a single file. Applies to all files

![image](https://user-images.githubusercontent.com/24353136/149084627-d6f6e464-6b01-4046-b3a8-d6137f61c59e.png)

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.py:
```diff
@@ -67,6 +76,34 @@ def validate_result_values(self):
 						item.idx, frappe.bold(item.lab_test_particulars)), title=_('Mandatory Results'))
 
+	@frappe.whitelist()
+	def modify_nursing_tasks(self, nursing_tasks):
```
This should first save the document before creating the tasks:

![image](https://user-images.githubusercontent.com/24353136/149085438-a9a4e3b8-f829-4828-abbd-8c595978f44e.png)

ruchamahabal replied:
Now that this is in the picture some issues I see here:
- users might miss this button during creation. We should allow it after saving too, delete the task if it's modified.
- There is no version log maintained at the bottom of the doc since this modification is stateless. Should we insert one for context?

akurungadam replied:
Not too sure if we should keep the `Modify Nursing Tasks`. Maybe bring this in based on request / feedback?

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.py:
```diff
@@ -67,6 +76,34 @@ def validate_result_values(self):
+			'Nursing Checklist Template Task',
+			filters=self.get_nursing_task_filters(self.template),
+			fields=['name', 'activity', 'mandatory', 'task_doctype'],
+		)
```
```suggestion
		template_tasks = frappe.get_all(
			'Nursing Checklist Template Task',
			filters=self.get_nursing_task_filters(self.template),
			fields=['name', 'activity', 'mandatory', 'task_doctype'],
		)
```

We should not use get_list on the child table since it checks for perms.

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.py:
```diff
@@ -67,6 +76,34 @@ def validate_result_values(self):
+		existing_tasks = frappe.get_list(
+			'Nursing Task',
+			filters={'reference_name': self.name, 'docstatus': 0},
+		)
```
```suggestion
		existing_tasks = frappe.get_all(
			'Nursing Task',
			filters={'reference_name': self.name, 'docstatus': 0},
		)
```

Again since this is happening in the backend safer to use `get_all`? The user will get unnecessary perm error.

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.py:
```diff
@@ -67,6 +76,34 @@ def validate_result_values(self):
+
+		if not existing_tasks:
+			NursingTask.create_nursing_tasks('Lab Test', self.name, template_tasks)
+			return
```
Why is this needed? Creation should always happen as per selection right, which is happening in lines 98-99? Irrespective of whether any task exists or not.

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.json:
```diff
@@ -555,14 +556,21 @@
+   "fetch_from": "template.nursing_checklist_template",
+   "fieldname": "modify_nursing_tasks",
+   "fieldtype": "Button",
+   "label": "Modify Nursing Tasks "
```
Any reason for keeping this button only in Lab Test?

ruchamahabal commented on healthcare/healthcare/doctype/clinical_procedure/clinical_procedure.js:
```diff
@@ -131,6 +131,10 @@ frappe.ui.form.on('Clinical Procedure', {
+
+		frm.add_custom_button(__("Nursing Tasks"), function () {
+				frappe.set_route("List", "Nursing Task", {"reference_name": frm.doc.name});
+		});
```
This button should not be shown before saving:
Also, club it under a parent button View -> Nursing Tasks. Since just "Nursing Tasks" is ambiguous.

ruchamahabal commented on healthcare/healthcare/doctype/clinical_procedure/clinical_procedure.py:
```diff
@@ -45,6 +47,17 @@ def after_insert(self):
+			for template in templates:
+				if not template:
+					continue
+				NursingTask.create_nursing_tasks_from_template(template, 'Clinical Procedure', self.name)
```
```suggestion
			cp_templates = frappe.db.get_value('Clinical Procedure Template', self.procedure_template, ['pre_op_nursing_checklist_template', 'post_op_nursing_checklist_template'], as_dict=1)
			NursingTask.create_nursing_tasks_from_template(cp_templates.pre_op_nursing_checklist_template, 'Clinical Procedure', self.name)
			NursingTask.create_nursing_tasks_from_template(cp.templates.post_op_nursing_checklist_template, 'Clinical Procedure', self.name)
```

- We can avoid `get_doc` here. `get_doc` is always expensive than `db.get_value`.
- Unnecessary for loop. Just call the method twice.

ruchamahabal commented on healthcare/healthcare/doctype/healthcare.py:
```diff
@@ -1,6 +1,12 @@
 	return []
+
+
+class PatientDocument(Document):
```
What's this for?

akurungadam replied:
Removed this for now, Anand you'll add back as needed?

ruchamahabal commented on healthcare/healthcare/doctype/healthcare_settings/healthcare_settings.js:
```diff
@@ -56,6 +56,11 @@ frappe.tour['Healthcare Settings'] = [
+	{
+		fieldname: 'validate_nursing_checklists',
+		title: __('validate_nursing_checklists'),
+		description: __('Checking this will automatically create a Sales Invoice whenever an appointment is booked for a Patient.')
```
Check Description

ruchamahabal commented on healthcare/healthcare/doctype/inpatient_record/inpatient_record.js:
```diff
@@ -34,6 +34,10 @@ frappe.ui.form.on('Inpatient Record', {
+
+		frm.add_custom_button(__("Nursing Tasks"), function () {
+				frappe.set_route("List", "Nursing Task", {"reference_name": frm.doc.name});
+		});
```
This button should not be shown before saving:
Also, club it under a parent button View -> Nursing Tasks. Since just "Nursing Tasks" is ambiguous.

ruchamahabal commented on healthcare/healthcare/doctype/inpatient_record/inpatient_record.json:
```diff
@@ -408,14 +410,27 @@
+  {
+   "fieldname": "admission_nursing_checklist_template",
+   "fieldtype": "Link",
+   "label": "Nursing Checklist Template",
```
```suggestion
   "label": "Admission Nursing Checklist Template",
```

ruchamahabal commented on healthcare/healthcare/doctype/lab_test/lab_test.py:
```diff
@@ -67,6 +76,34 @@ def validate_result_values(self):
+	@frappe.whitelist()
+	def get_nursing_task_filters(self, lab_test_template):
+		lab_test_template = frappe.get_doc('Lab Test Template', lab_test_template)
+		filters = {'parent': lab_test_template.nursing_checklist_template}
```
```suggestion
		nursing_checklist_template = frappe.db.get_value('Lab Test Template', lab_test_template, 'nursing_checklist_template')
		filters = {'parent': nursing_checklist_template}
```

Also, do we need this whitelisted method? It's just called from a single function, just compute it there itself.

ruchamahabal commented on healthcare/healthcare/doctype/nursing_checklist_template/nursing_checklist_template.json:
```diff
@@ -0,0 +1,65 @@
+   "fieldname": "department",
+   "fieldtype": "Link",
+   "label": "Department",
+   "options": "Medical Department"
```
Add Department to List View and Filter View:

![image](https://user-images.githubusercontent.com/24353136/149090074-ce487689-c435-4855-8e61-71d9a424c3c6.png)

ruchamahabal commented on healthcare/healthcare/doctype/nursing_task/nursing_task.json:
```diff
@@ -0,0 +1,157 @@
+ ],
+ "sort_field": "modified",
+ "sort_order": "DESC",
+ "track_changes": 1
```
![image](https://user-images.githubusercontent.com/24353136/149090381-e78f314f-5e2d-43f4-8496-c14489901124.png)

- Set title for this doctype, maybe activity
- Add more fields to List View and Filter

akurungadam replied:
![Screen Shot 2022-01-18 at 1 26 55 PM](https://user-images.githubusercontent.com/3326959/149894531-e734cae4-b240-43b0-8bdf-35967c0eb0f0.png)

ruchamahabal commented on healthcare/healthcare/doctype/nursing_task/nursing_task.json:
```diff
@@ -0,0 +1,157 @@
+   "in_list_view": 1,
+   "label": "Status",
+   "options": "Requested\nDraft\nReceived",
+   "reqd": 1
```
Instead of these in reference to [https://www.hl7.org/fhir/valueset-task-status.html](https://www.hl7.org/fhir/valueset-task-status.html) should we have:

- Draft
- Cancelled
- Requested
- In Progress
- On Hold
- Failed
- Completed

akurungadam replied:
added some more optional statuses (Ready, Approved, Rejected, Entered in Error etc.)

ruchamahabal commented on healthcare/healthcare/utils.py:
```diff
@@ -814,4 +814,18 @@ def before_tests():
+def validate_nursing_tasks(document):
+	healthcare_settings = frappe.get_single("Healthcare Settings")
+	if not healthcare_settings.validate_nursing_checklists:
+		return True
```
```suggestion
	if not frappe.db.get_single_value("Healthcare Settings", "validate_nursing_checklists")
		return True
```

ruchamahabal commented on healthcare/healthcare/utils.py:
```diff
@@ -814,4 +814,18 @@ def before_tests():
+	if not tasks:
+		return True
+
+	frappe.throw('Please complete related nursing tasks before submission')
```
```suggestion
	frappe.throw(_('Please complete linked Nursing Tasks before submission'))
```

ruchamahabal replied:
Also, what's the purpose of marking nursing tasks as Mandatory? We should ignore validating non-mandatory tasks here. Also, while making the checklist, let mandatory be set by default. User can uncheck for whichever tasks they don't want mandatory.

ruchamahabal replied:
Add this mandatory check to the multi-select dialog as well

akurungadam replied:
Checking for 'Completed' status now, Nursing Tasks are now submitted on status 'Requested'

ruchamahabal commented on healthcare/healthcare/doctype/nursing_task/nursing_task.json:
```diff
@@ -0,0 +1,157 @@
+  {
+   "fieldname": "description",
+   "fieldtype": "Data",
+   "label": "Description"
```
- The description should be read-only and the same data type as in the checklist/activity. Set it from Checklist desc
- **Mandatory** checkbox should be **Set Only Once**?
- Add role field in Nursing Task since we set it in the template. If Role is set, apply a filter on the user field for users with that role else keep it open-ended.
- Rearrange fields
	- LHS: Activity, User, Description
	- RHS: Status, From Date, To Date
- Reference Doctype --> Reference Document Type
- Task DocType --> Task Document Type

![image](https://user-images.githubusercontent.com/24353136/149093801-105b9115-cb54-481a-a1c3-8775a51cc55b.png)

The whole task reference concept is not very evident from the design. Need to rethink the purpose, utility, and usage. Should be easy for the user to understand.

ruchamahabal commented on healthcare/healthcare/doctype/therapy_plan/therapy_plan.py:
```diff
@@ -48,6 +53,14 @@ def set_therapy_details_from_template(self):
 
+	def after_insert(self):
+		if self.therapy_plan_template:
+			therapy_template = frappe.get_doc('Therapy Plan Template', self.therapy_plan_template)
```
Replace with `get_value` as pointed out in the comments earlier.

ruchamahabal commented on healthcare/healthcare/doctype/nursing_task/nursing_task.py:
```diff
@@ -0,0 +1,42 @@
+class NursingTask(Document):
+	@classmethod
+	def create_nursing_tasks_from_template(cls, template_name=None, dt=None, dn=None, append=False):
+		tasks = frappe.get_list(
```
```suggestion
		tasks = frappe.get_all(
```

ruchamahabal commented on healthcare/healthcare/doctype/nursing_task/nursing_task.py:
```diff
@@ -0,0 +1,42 @@
+				'reference_doctype': dt,
+				'reference_name': dn,
+			}
+			existing_tasks = frappe.get_list('Nursing Task', filters=filters)
```
```suggestion
			existing_tasks = frappe.get_all('Nursing Task', filters=filters)
```

ruchamahabal commented on healthcare/healthcare/doctype/nursing_task/test_nursing_task.py:
```diff
@@ -0,0 +1,42 @@
+		self.lab_template.save()
+		create_lab_test(self.lab_template)
+
+		self.assertEqual(frappe.db.count('Nursing Task'), task_count + 1)
```
Increase test coverage:

- Modifying checklists
- Discharge / Pre op / Post op / Admission checklists working properly

ruchamahabal commented on healthcare/healthcare/doctype/therapy_session/therapy_session.py:
```diff
@@ -58,6 +61,14 @@ def on_cancel(self):
 		self.update_sessions_count_in_therapy_plan(on_cancel=True)
 
+	def after_insert(self):
+		therapy_plan = frappe.get_doc('Therapy Plan', self.therapy_plan)
+		if therapy_plan.therapy_plan_template:
+			plan_template = frappe.get_doc('Therapy Plan Template', therapy_plan.therapy_plan_template)
+			template = plan_template.nursing_checklist_template
```
For therapy sessions, checklists should be linked to Therapy Type instead of Plan. Also, avoid `get_doc`

ruchamahabal commented on healthcare/healthcare/doctype/nursing_task/nursing_task.py:
```diff
@@ -0,0 +1,42 @@
+	def create_nursing_tasks(cls, dt=None, dn=None, tasks=None):
+		for task in tasks:
+			options = {
+				'doctype': 'Nursing Task',
+				'status': 'Requested',
+				'reference_doctype': dt,
+				'reference_name': dn,
+				'activity': task.activity,
+				'mandatory': task.mandatory,
+				'task_doctype': task.task_doctype,
```
No mapping of dates from template to task?

The From Date should be set relative to Reference Doctype's transaction date. And End Date should be calculated as From Date + Duration

Check how [Onboarding Template](https://docs.erpnext.com/docs/v13/user/manual/en/human-resources/employee-onboarding#31-employee-onboarding-template) works in ERPNext.

akurungadam replied:
Added a time_offset field to set the requested_start_time

ruchamahabal commented:
General Pointers:

- Title/Naming for all doctypes
- List View/Filter defaults for all doctypes
- Document States and color mapping
- Default Roles for all doctypes (currently all are just accessible to System Managers)
- Connections/Document Links should be configured for all doctypes for easy navigation and user context
- Try to keep uniform quote styles everywhere or at least throughout a single file.
- Mixed tabs/spaces in some files. Stick to tabs as per our current codebase.
- Check search fields for every doctype
- IMO, since this is applicable for every type of template, tests should cover it for all doctypes: Clinical Procedure, Admission, Discharge, Therapy Session, and not just Lab Test

akurungadam committed:
fix: added defaults for healthcare activity
add statuses to nursing tasks as in fhir resource spec
added validations to nursing task
option to change status in nursing task
other minor fixes

akurungadam committed:
fix: changed nursing task creation -
clinical procedure on submit preop tasks and on complete postop tasks
inpatient record on admit preop tasks and on schedule discharge postop tasks
moved template link and tasks creation to therapy type and therapy session

add linked doctypes in all doctypes linking to nursing task
removed inpatient_record_dashboard.py

validate nursing tasks now checks for completed status instead of sumit
validate nursing tasks only for preop tasks

akurungadam committed:
fix: remove nursing tasks custom button
other fixes as suggested in pr comments

akurungadam committed:
Update healthcare/healthcare/doctype/inpatient_record/inpatient_record.json

Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>

akurungadam committed:
fix: nursing tasks tests

akurungadam committed:
fix: list view indicators for Nursing Task

akurungadam committed:
feat: option to create nursing tasks from encounter
fix: nursing task task_document description, set mandatroy
fix: set nursing task status to Requested on_submit
api to create nursing tasks based on template, minor fixes, code cleanup

ChillarAnand committed:
chore: Fix lint issues

ChillarAnand committed:
test: Fix failing tests

ChillarAnand committed:
test: Add tests for lab test submission

ChillarAnand committed:
test: Add nursing tasks tests for healthcare services

ChillarAnand committed:
Merge branch 'develop' into nursing-tasks

ChillarAnand committed:
fix: Add missing fields in clinical procedure template

ChillarAnand committed:
fix: Fix broken tests

ChillarAnand committed:
fix: Fix frappe test case import

ruchamahabal committed:
Merge branch 'develop' into nursing-tasks

ruchamahabal committed:
fix: translation, field fixes

ruchamahabal committed:
chore: convert space to tabs for uniform formatting(utils.py)

ruchamahabal committed:
chore: ignore formatting changes in blame

ruchamahabal committed:
fix: remove Modify Nursing Tasks flow

ruchamahabal committed:
fix(navigation): Add nursing module doctypes to workspace
